controllers/openstackdataplane_controller.go

```diff
@@ -227,176 +252,58 @@ func (r *OpenStackDataPlaneReconciler) Reconcile(ctx context.Context, req ctrl.R
-		instance.Status.Conditions.Set(condition.FalseCondition(condition.DeploymentReadyCondition, condition.NotRequestedReason, condition.SeverityInfo, condition.DeploymentReadyInitMessage))
-	}
-
 	return ctrl.Result{}, nil
 }
 
 // SetupWithManager sets up the controller with the Manager.
 func (r *OpenStackDataPlaneReconciler) SetupWithManager(mgr ctrl.Manager) error {
 	return ctrl.NewControllerManagedBy(mgr).
 		For(&dataplanev1.OpenStackDataPlane{}).
-		Owns(&dataplanev1.OpenStackDataPlaneNode{}).
-		Owns(&dataplanev1.OpenStackDataPlaneRole{}).
 		Complete(r)
 }
 
-// createOrPatchDataPlaneResources -
-func createOrPatchDataPlaneResources(ctx context.Context, instance *dataplanev1.OpenStackDataPlane, helper *helper.Helper) (ctrl.Result, error) {
-	// create DataPlaneRoles
-	roleManagedHostMap := make(map[string]map[string]baremetalv1.InstanceSpec)
-	err := createOrPatchDataPlaneRoles(ctx, instance, helper, roleManagedHostMap)
-	if err != nil {
-		instance.Status.Conditions.MarkFalse(
-			condition.ReadyCondition,
-			condition.ErrorReason,
-			condition.SeverityError,
-			dataplanev1.DataPlaneErrorMessage,
-			err.Error())
-		return ctrl.Result{}, err
-	}
-
-	// Create DataPlaneNodes
-	err = createOrPatchDataPlaneNodes(ctx, instance, helper)
-	if err != nil {
-		instance.Status.Conditions.MarkFalse(
-			condition.ReadyCondition,
-			condition.ErrorReason,
-			condition.SeverityError,
-			dataplanev1.DataPlaneNodeErrorMessage,
-			err.Error())
-		return ctrl.Result{}, err
-	}
-
-	// Get All Nodes
-	nodes := &dataplanev1.OpenStackDataPlaneNodeList{}
-	listOpts := []client.ListOption{
-		client.InNamespace(instance.GetNamespace()),
-	}
-
-	err = helper.GetClient().List(ctx, nodes, listOpts...)
-	if err != nil {
-		return ctrl.Result{}, err
-	}
+func (a *DataPlaneAnnotations) updateNodeSetAnnotations(ctx context.Context, h *helper.Helper, instance *dataplanev1.OpenStackDataPlaneNodeSet) (err error) {
+	retryErr := retry.RetryOnConflict(retry.DefaultRetry, func() error {
+		// If we fail and need to retry, we should re-check the object to obtain the currentAnnotations
+		// since they may have changed.
+		currentAnnotations := instance.ObjectMeta.GetAnnotations()
 
-	if len(nodes.Items) < len(instance.Spec.Nodes) {
-		util.LogForObject(helper, "All nodes not yet created, requeueing", instance)
-		return ctrl.Result{RequeueAfter: time.Second * 10}, nil
-	}
+		// Merge the desired and current label maps
+		mergedAnnotations := make(map[string]string)
+		for k, v := range currentAnnotations {
+			mergedAnnotations[k] = v
+		}
+		for k, v := range a.Annotations {
+			mergedAnnotations[k] = v
+		}
 
-	// Order the nodes based on Name
-	sort.SliceStable(nodes.Items, func(i, j int) bool {
-		return nodes.Items[i].Name < nodes.Items[j].Name
+		instance.ObjectMeta.SetAnnotations(mergedAnnotations)
+		updateErr := h.PatchInstance(ctx, instance)
+		return updateErr
 	})
 
-	err = deployment.BuildBMHHostMap(ctx, helper, instance, nodes, roleManagedHostMap)
```

Comment:
Hey @rabi Thanks for the insights. I'll touch base with you today to make sure I get this right. I haven't tested with baremetal provisioning yet, everything has be pre-provisioned so far. So this feedback is super valuable.

I think we want to handle all of the baremetal provisioning from the nodeSet controller and just have the OpenStackDataPlane as a trigger for the Ansible configuration.

Reply:
>  I'll touch base with you today to make sure I get this right.
sure
> I think we want to handle all of the baremetal provisioning from the nodeSet controller 

That's where it is done now was well (s/role/nodeset). The only thing dataplane doing is creating/patching the roles/nodes atm. Having a dataplane CRD/controller to just add/update annotations on set of nodesets does not sound right to me. Either we've the dataplane CR which owns the nodeset(s) and the manages them in it's reconciliation or we don't have it.

Reply:
I do think we can move forward with this PR with the scope of:

1. rename Role to NodeSet
2. inline NodeSet names only on DataPlane
3. remove Node CRD/Controller

Agree we can probably remove DataPlane next. I'd like to see some progress on this PR though. It feels stalled to me b/c it's trying to do too much. Let's iterate on what we change. If we want to drop (2) from above to make it even smaller scope, that's fine. But let's not block on trying to make this PR do more.

Reply:
Yeah, agree to push that discussion. S, I think we have currently achieved that scope right? From my side, I need to test that this works with Baremetal deployments, because I have focused on pre-provisioned to this point.

This particular comment, I probably need to address in the nodeSet controller when we create the baremetal nodes. This seems to be adding configuration to the baremetal instance, so we'll need to preserve this elsewhere if we remove it from OpenStackDataplane